tests/order/test_base_order_utils.cairo

```diff
@@ -1,51 +1,229 @@
+    assert(is_decrease_order(OrderType::LimitDecrease), 'invalid limit dec. res');
+    assert(is_decrease_order(OrderType::StopLossDecrease), 'invalid stop loss res');
+    assert(is_decrease_order(OrderType::Liquidation), 'invalid liquidation res');
+
+    // Test other orders
+    assert(!is_decrease_order(OrderType::MarketIncrease), 'invalid market inc. res');
+    assert(!is_decrease_order(OrderType::LimitIncrease), 'invalid limit inc. res');
+}
+
+#[test]
+fn given_normal_conditions_when_is_liquidation_order_then_works() {
+    // Test position orders
+    assert(is_liquidation_order(OrderType::Liquidation), 'invalid liquidation inc. res');
+    // Test other orders
+    assert(!is_liquidation_order(OrderType::MarketDecrease), 'invalid market dec. res');
+    assert(!is_liquidation_order(OrderType::MarketSwap), 'invalid market swap res ');
 }
 
+
 #[test]
 fn given_normal_conditions_when_validate_order_trigger_price_then_works() {
-    // TODO when oracle
-    // let oracle_address: ContractAddress = 'oracle'.try_into().unwrap();
-    // start_mock_call(oracle_address, 'get_primary_price', Price { min: 9, max: 11 });
-    // let oracle = IOracleSafeDispatcher { contract_address: oracle_address };
-    // validate_order_trigger_price(
-    //     oracle,
-    //     index_token: 'token'.try_into().unwrap(),
-    //     order_type: OrderType::LimitIncrease,
-    //     trigger_price: 10,
-    //     is_long: true,
-    // );
-    // stop_mock_call(oracle_address, 'get_primary_price');
-    assert(true, 'Tautology');
+    // Setup
+    let (_, _, _, oracle) = setup();
+    let index_token = contract_address_const::<'ETH'>();
+    let price = Price { min: 100000, max: 200000 };
+    oracle.set_primary_price(index_token, price);
+
+    // Test
+
+    // Test swap orders validates
+    validate_order_trigger_price(oracle, index_token, OrderType::MarketSwap, 100, true);
+
+    // Test limit increase orders
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::LimitIncrease, price.max + 1, true
+    );
+
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::LimitIncrease, price.min - 1, false
+    );
+
+    // Test limit decrease orders
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::LimitDecrease, price.min - 1, true
+    );
+
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::LimitDecrease, price.max + 1, false
+    );
+
+    // Test stop loss orders
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::StopLossDecrease, price.min + 1, true
+    );
+
+    validate_order_trigger_price(
+        oracle, index_token, OrderType::StopLossDecrease, price.max - 1, false
+    );
+
+    assert(true, 'e');
+}
+
+#[test]
+#[should_panic(
+    expected: ('invalid_order_price', 100000, 200000, 199999, 6053968548023263173723725853541)
+)]
+fn given_limit_increase_price_higher_than_trigger_when_validate_order_trigger_price_then_works() {
```

Comment:
Some functions are named `...then_works` but it should be `...then_panic`

Reply:
sorry, I forgot about that. I will fix it

Reply:
fixed here
https://github.com/tevrat-aksoy/satoru/commit/b592da250dee903e5ccbfea38383ccebf9e98dd3